health/src/main/java/com/health/operations/Chunk.java

```diff
@@ -0,0 +1,217 @@
+
+		// fill the table
+		Record tmp = new Record(table);
+
+		tmp.setValue("date", 10.0);
+		tmp.setValue("meetwaarde1", 8.0);
+		tmp.setValue("name", "Piet");
+		tmp.setValue("meetwaarde2", 20.0);
+
+		tmp = new Record(table);
+		tmp.setValue("date", 10.0);
+		tmp.setValue("meetwaarde1", 10.0);
+		tmp.setValue("name", "Hein");
+		tmp.setValue("meetwaarde2", 10.0);
+
+		tmp = new Record(table);
+		tmp.setValue("date", 10.0);
+		tmp.setValue("meetwaarde1", 10.0);
+		tmp.setValue("name", "Dolf");
+		tmp.setValue("meetwaarde2", -1.0);
+
+		tmp = new Record(table);
+		tmp.setValue("date", 10.0);
+		tmp.setValue("meetwaarde1", 10.0);
+		tmp.setValue("name", "Piet");
+		tmp.setValue("meetwaarde2", 10.0);
+
+		tmp = new Record(table);
+		tmp.setValue("date", 10.0);
+		tmp.setValue("meetwaarde1", 10.0);
+		tmp.setValue("name", "Piet");
+		tmp.setValue("meetwaarde2", 3.0);
+
+		tmp = new Record(table);
+		tmp.setValue("date", 10.0);
+		tmp.setValue("meetwaarde1", 10.0);
+		tmp.setValue("name", "Dolf");
+		tmp.setValue("meetwaarde2", 10.0);
+
+		DateTimeFormatter formatter = DateTimeFormatter
+				.ofPattern("d[d]/M/yyyy");
+		LocalDate day1 = LocalDate.parse("10/2/2013", formatter);
+		System.out.println(day1.toString());
+		Period per = Period.ofDays(1);
+		LocalDate day2 = day1.plus(per);
+
+		Map<String, AggregateFunctions> operations = new HashMap<String, AggregateFunctions>();
+
+		operations.put("meetwaarde1", AggregateFunctions.Average);
+		operations.put("meetwaarde2", AggregateFunctions.Min);
+
+		Table chunkedTable = chunkByString(table, "name", operations);
+
+		System.out.println(Output.formatTable(chunkedTable));
+
+	}
+*/
+	/**
+	 * A function to chunk a dataSet by time.
+	 *
+	 * @param table
+	 *            the Table to be chunked.
+	 * @param column
+	 *            the column on which to chunk should be a column of type Date.
+	 * @param operations
+	 *            a map of columns and their aggreagate operation.
+	 * @param period
+	 *            the period between chunk, could be days, months, years.
+	 * @return
+	 */
+	public static Table chunkByTime(final Table table, final String column,
+			final Map<String, AggregateFunctions> operations,
+			final Period period) {
+
+		Table chunkedTable = new Table(table.getColumns());
+
+		// find first date
+
+		// new date tmp = first date + period.
+
+		// make record[] for all record where date < tmp && date >= first date
+
+		// aggregate the records
+
+		// add record to chunkedTable
+
+		// increment dates and loop back
+
+		return chunkedTable;
+	}
+
+	/**
+	 * chunks the data on same string, in the given column.
+	 * 
+	 * @param table
+	 * @param operations
+	 *            a map of columns and their aggreagate operation.
+	 * @param column
+	 *            on which the data is chunked with the same string.
+	 * @return
+	 */
+	public static Table chunkByString(final Table table, final String column,
+			final Map<String, AggregateFunctions> operations) {
```

Comment:
Wouldn't it be better to subdivide this large method into several smaller methods?